### PR TITLE
Add pull request mode

### LINKED ISSUE #1042: Open pull request instead of directly pushing changes
Hi, I really like this action, but it lacks one key feature: you should be able to open a pull request using this action which can be reviewed by project maintainers to ensure code security. If the formatter messes things up, so changes don't get pushed to production automatically, and therefore supply chain attacks with manipulated formatting scripts would be way harder to perform.

Dependabot does this on every change, kindly asking to merge the PR instead of directly committing to the code base.

### PR DESCRIPTION
**Pull request by Le0X8**
Closes #1042.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,6 +67,38 @@ Please note that this mode requires:
...
+
+name: Rustfmt
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+    permissions: write-all
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@nightly
```
Comment (mbrobbel) [inline README.md:93]: I guess this should work with a stable toolchain?
```suggestion
      - uses: dtolnay/rust-toolchain@stable
```

**Review by mbrobbel — COMMENTED** [review]
This looks great, thank you!

Does this also work when the PR is created from a fork? If not, maybe we should add that to the README.

Can you also bump `two` to `three` here:
https://github.com/mbrobbel/rustfmt-check/blob/5a0f1b2b7ed39f4ff7e5af46c2cb3e9c0f496dbc/README.md?plain=1#L9

**Comment by Le0X8** [comment]
> Does this also work when the PR is created from a fork? If not, maybe we should add that to the README.

Yes, because the action runs on every commit (pull request merging counts as a commit too).

**Comment by Le0X8** [comment]
@mbrobbel Everything should be fixed now. Full write access is required to create branches and push changes to them.

**Comment by mbrobbel** [comment]
> @mbrobbel Everything should be fixed now. Full write access is required to create branches and push changes to them.

Sorry, I meant specifying only [the required scopes](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#defining-access-for-the-github_token-scopes) instead of using `write-all`. 

I think you need `contents: write` and `pull-requests: write`. But I'm not sure so this needs some testing.

**Comment by Le0X8** [comment]
@mbrobbel Yes, you're right, these permissions work as shown [here](https://github.com/Le0X8/playground/commit/5fc4d9e447e7cd6c7684240b2975983b398d980c).

**Review by mbrobbel — APPROVED** [review]
Thank you for this awesome contribution!

**Pull request merged**